src/probnum/quad/bq_methods/bayesian_quadrature.py

```diff
@@ -22,104 +30,244 @@ class BayesianQuadrature:
 
     Parameters
     ----------
-    kernel:
+    kernel :
         The kernel used for the GP model.
-    policy:
+    measure :
+        The integration measure.
+    policy :
         The policy for acquiring nodes for function evaluations.
+    belief_update :
+        The inference method.
+    stopping_criteria :
+        List of criteria that determine convergence.
     """
 
-    def __init__(self, kernel: Kernel, policy: Callable) -> None:
+    def __init__(
+        self,
+        kernel: Kernel,
+        measure: IntegrationMeasure,
+        policy: Policy,
+        belief_update: BQBeliefUpdate,
+        stopping_criteria: List[StoppingCriterion],
+    ) -> None:
         self.kernel = kernel
+        self.measure = measure
         self.policy = policy
+        self.belief_update = belief_update
+        self.stopping_criteria = stopping_criteria
 
     @classmethod
-    def from_interface(
+    def from_bayesquad(
```

Comment:
I recommend renaming this (e.g. `from_problem`), since it is not only called from the interface, but also as a convenience for a more experienced user compiling a quadrature method. You might also want to pass the integrand here to determine differentiability or else in the future.

Reply:
Renamed :white_check_mark: 
Didn't we discuss at some point that the function should not be an argument of `BayesianQuadrature`, so that one instance of `BayesianQuadrature` can in principle integrate multiple functions? Maybe I don't fully understand your design choice with the classmethod there :thinking:

Reply:
Correct the `__init__` method should not require the function, but only the prior information. 

However, you want to support (at least one, possibly multiple) constructors which define standard ways to instantiate a BQ method from the information you have available. For example if you have a function which is made up from basic differentiable functions, your `classmethod` can then define default prior choices. You can also use this to construct a BQ method with an example function which has properties shared across all the functions you want to integrate with that method.

Reply:
That sounds reasonable once we impose structure on how the integrand shall be passed (alongside additional information about the function). For now this is not used and the function is not needed in the constructor. Does it make sense to address this at a later point? Otherwise there will be an unused argument `fun`.